ledger-api/rs-grpc-bridge/src/main/java/com/daml/grpc/adapter/server/rs/ServerSubscriber.java

```diff
@@ -113,8 +113,13 @@ public void onError(Throwable throwable) {
     executionSequencer.sequence(
         () -> {
           if (!responseObserver.isCancelled()) {
-            responseObserver.onError(throwable);
-            completionPromise.completeExceptionally(throwable);
+            Throwable newThrowable = translateThrowableInOnError(throwable);
+            responseObserver.onError(newThrowable);
+            // TODO error codes: Complete with thrownable or newThrowable?
+            //                   This promise seems to to used only to fail materialized
+            // Future[Unit] in streaming endpoints.
+            //                   Is that materialized Future[Unit] used anywhere anyway?
```

Comment:
I don't understand this comment.

Reply:
I was considering what `completionPromise.completeExceptionally` is used for. Afaict: that promise has a corresponding future that is used as a materialized value of the stream which in turn ?isn't used anywhere?